stl/inc/future

```diff
@@ -1266,7 +1314,10 @@ public:
     packaged_task() = default;
 
     template <class _Fty2, enable_if_t<!is_same_v<_Remove_cvref_t<_Fty2>, packaged_task>, int> = 0>
-    explicit packaged_task(_Fty2&& _Fnarg) : _MyPromise(new _MyStateType(_STD forward<_Fty2>(_Fnarg))) {}
+    explicit packaged_task(_Fty2&& _Fnarg) : _MyPromise(new _MyStateType(_STD forward<_Fty2>(_Fnarg))) {
+        static_assert(_Is_invocable_r<_Ret, _Fty2&, _ArgTypes...>::value,
+            "The function object must be callable with _ArgTypes... and return _Ret (N4988 [futures.task.members]/3).");

```

Comment:
Now the change is exactly reflecting [[futures.task.members]/3](https://eel.is/c++draft/futures.task.members#3), but the _Mandates_ doesn't seem very correct when `remove_reference_t<_Fty2>` is cv-qualified.
Should we use `decay_t<_Fty2>&` or `_Remove_cvref_t<_Fty2>&`? I've mailed to LWG Chair to submit an LWG issue to say `decay_t` in [futures.task.members]/3.

Reply:
[LWG4154](https://cplusplus.github.io/LWG/issue4154)